ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/OfferingToStateMapper.kt

```diff
@@ -0,0 +1,21 @@
+package com.revenuecat.purchases.ui.revenuecatui
+
+import com.revenuecat.purchases.Offering
+
+@Suppress("ReturnCount")
+internal fun Offering.toPaywallViewState(): PaywallViewState {
+    val paywallData = this.paywall
+        ?: return PaywallViewState.Error("No paywall data for offering: $identifier")
+    val packageById = this.availablePackages.associateBy { it.identifier }
+    val packages = paywallData.config.packages.map { packageId ->
+        packageById[packageId] ?: return PaywallViewState.Error("No package $packageId in offering $identifier")
+    }
+    return PaywallViewState.Template2(
+        paywallData = paywallData,
+        packages = packages,
+        selectedPackage = packages.first(), // TODO-PAYWALLS: Use correct default package
+        shouldShowRestorePurchasesButton = paywallData.config.displayRestorePurchases,
+        shouldShowTermsButton = paywallData.config.termsOfServiceURL != null,
+        shouldShowPrivacyPolicyButton = paywallData.config.privacyURL != null,
```

Comment:
I wonder if these are needed here. The state already has `PaywallData` so the future `Footer` "compoment"/"view" whatever can derive it from its config?

Reply:
Right, it could. I mostly wanted to abstract as much logic as possible away from the composables. Ideally, the state has everything it needs to render without making any decisions (we are not there yet)